packages/itwinui-react/src/utils/hooks/useControlledState.ts

```diff
@@ -27,14 +30,40 @@ export const useControlledState = <T>(
     () => (controlledState !== undefined ? controlledState : uncontrolledState),
     [controlledState, uncontrolledState],
   );
+  const oldState = React.useRef<T>(state);
 
   const setState = React.useCallback(
     (value) => {
+      const newValue = (() => {
+        if (typeof value === 'function') {
+          return (value as (prevState: T) => T)(state);
+        }
+        return value;
+      })();
```

Comment:
This looks unrelated to the main change. Do we even support passing a function in any of the `on[…]Change` props?

Reply:
We can change the type as this is an internal hook. The `setIsExpanded` call in `SideNavigation` is also not exposed to consumers, so we can change it as well. After that, we can remove this entirely `newValue` declaration.